File: Daily Test/Java/Bit Mask Matching 12.6.2025.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String s = sc.next();
        for (char c : s.toCharArray()) {
            int idx = c - 'A';
            if (((n >> idx) & 1) == 0) {
                System.out.print("NO");
                return;
            }
        }
        System.out.print("YES");
    }
}

File: Daily Test/C/Bit Mask Matching 12.6.2025.c
#include <stdio.h>
#include <string.h>

int main() {
    unsigned int n;
    char s[101];
    scanf("%u %s", &n, s);
    for (int i = 0; s[i]; i++) {
        int idx = s[i] - 'A';
        if (((n >> idx) & 1) == 0) {
            printf("NO");
            return 0;
        }
    }
    printf("YES");
    return 0;
}